providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/grpc/GrpcConnector.java

```diff
@@ -0,0 +1,173 @@
+
+import dev.openfeature.contrib.providers.flagd.FlagdCache;
+import dev.openfeature.contrib.providers.flagd.FlagdOptions;
+import dev.openfeature.flagd.grpc.Schema;
+import dev.openfeature.flagd.grpc.ServiceGrpc;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.ProviderState;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.netty.channel.epoll.EpollDomainSocketChannel;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.unix.DomainSocketAddress;
+import io.netty.handler.ssl.SslContextBuilder;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.net.ssl.SSLException;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+
+@Slf4j
+public class GrpcConnector {
+
+    private final ServiceGrpc.ServiceBlockingStub serviceBlockingStub;
+    private final ServiceGrpc.ServiceStub serviceStub;
+    private final ManagedChannel channel;
+    private final int maxEventStreamRetries;
+
+    private int eventStreamAttempt = 1;
+    private int eventStreamRetryBackoff;
+    private final int startEventStreamRetryBackoff;
+    private final long deadline;
+
+    private final FlagdCache cache;
+    private final Consumer<ProviderState> setState;
+
+    public GrpcConnector(final FlagdOptions options, final FlagdCache cache, Consumer<ProviderState> setState) {
+        this.channel = nettyChannel(options);
+        this.serviceStub = ServiceGrpc.newStub(channel);
+        this.serviceBlockingStub = ServiceGrpc.newBlockingStub(channel);
+        this.maxEventStreamRetries = options.getMaxEventStreamRetries();
+        this.startEventStreamRetryBackoff = options.getRetryBackoffMs();
+        this.eventStreamRetryBackoff = this.startEventStreamRetryBackoff;
+        this.deadline = options.getDeadline();
+        this.cache = cache;
+        this.setState = setState;
+    }
+
+    public void initialize(EvaluationContext evaluationContext) throws RuntimeException {
+        try {
+            // try a dummy request
+            this.serviceBlockingStub
+                    .withWaitForReady()
+                    .withDeadlineAfter(this.deadline, TimeUnit.MILLISECONDS)
+                    .resolveBoolean(Schema.ResolveBooleanRequest.newBuilder().setFlagKey("ready?").build());
+        } catch (StatusRuntimeException e) {
+            // only return the exception if we don't meet the deadline
+            if (Status.DEADLINE_EXCEEDED.equals(e.getStatus())) {
+                throw e;
+            }
+        } finally {
+            // try in background to open the event stream
+            this.startEventStream();
+        }
+        /**
+         * Evaluate if the following makes sense
```

Comment:
I went through this code and IMO the initial dummy evaluation is a good enough strategy

I don't think we should fail in the init method. Another alternative is to use a timer and set error state if event stream connectivity fails

Reply:
> Another alternative is to fail the init method on purpose and let the status recover once we are connected

I could be misunderstanding, but if you throw in `init` it will cause the SDK to send an error event, which we don't want if things are functioning normally.